jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/JdbcSerDe.java

```diff
@@ -131,7 +131,8 @@ public void initialize(Configuration configuration, Properties tableProperties,
         row = new ArrayList<>(hiveColumnNames.length);
       }
     } catch (Exception e) {
-      throw new SerDeException("Caught exception while initializing the SqlSerDe", e);
+      log.error("Caught exception while initializing the SqlSerDe", e);
+      throw new SerDeException(e);
```

Comment:
> Log and rethrow is usually an anti pattern and most of the time it will lead into reporting the same problem multiple times.
> 
> Also it is usually a good thing to wrap and rethrow an exception with a message appropriate for the current abstraction.

Make sense. How about this change?
`throw new SerDeException("Caught exception while initializing the SqlSerDe: " + e.getMessage(), e);`

> I am not sure that the code here is responsible for "losing" the real exception. Have you checked higher or lower in the call hierarchy to see if there is something wrong going on there?

I'm sure that's the root of the problem. 
Method _**runQuery()**_  in SQLOperation.java will catch final exception and rethrow _**toSQLException**_  which show the root exception message to the client, and the exception messgae is retrieved from original exception in JdbcSerDe using _**e.getMessage()**_. howerver, current JdbcSerDe code always return exception message "**Caught exception while initializing the SqlSerDe**".
https://github.com/apache/hive/blob/a63f4b38c4ab9baf5d99f233fd3b49e8e52fb6c6/jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/JdbcSerDe.java#L133-L134
https://github.com/apache/hive/blob/a63f4b38c4ab9baf5d99f233fd3b49e8e52fb6c6/service/src/java/org/apache/hive/service/cli/operation/SQLOperation.java#L245-L246
https://github.com/apache/hive/blob/a63f4b38c4ab9baf5d99f233fd3b49e8e52fb6c6/service/src/java/org/apache/hive/service/cli/operation/Operation.java#L365-L368

Reply:
Actually, we can get all exception stack trace which contains all causes along with their messages from hiveserver2 log, and we can find partial real exception message from hiveserver2 log. 
But client user(beeline) can not get real exception message, because `CommandProcessorException` only return message  "**Caught exception while initializing the SqlSerDe**"

Reply:
It seems that this is a problem of the client (Beeline) and not a problem of the server (HS2).

A stacktrace can be composed of many nested exceptions one causing the other. It doesn't seem like a very good idea to me to just concatenate the messages when propagating the exception from one abstraction to the other.

```java
throw new SerDeException("Caught exception while initializing the SqlSerDe: " + e.getMessage(), e);
```
The snippet above will make the exception in the HS2 logs more verbose than necessary since it will essentially have duplicate information. 

Even the current version of the code contains duplicate information:
```java
throw new SerDeException("Caught exception while initializing the SqlSerDe", e);
```
The stacktrace shows that this exception is thrown from inside a `catch` block so mentioning "Caught exception while" does not add additional info but just states the obvious. Moreover, the stack trace also shows that we are inside the initialize method, in `JdbcSerDe` class so putting "initializing the SqlSerDe" is redundant and possibly misleading cause there is no `SqlSerDe` class.

What I want to highlight is that exceptions are for developers and not for end users. It is very important to provide to the end user meaningful error messages but we should be mindful on how we achieve this result.